Week1/homework/app.js

```diff
@@ -1,11 +1,170 @@
+  //     const li = document.createElement('li');
+  //     ul.appendChild(li);
+  //     li.innerText = bookTitles[i];
+  //   }
+  // }
+
+  /* cSpell:disable */
+
+  // 1.4 Make an object (not an array!) containing information for each book.
+  // Each property of this object should be another(i.e., nested) object with the book ID you thought up in step 1.1
+  // as a key, and at least the following properties: title, language and author.
+  const bookInformation = {
+    maze_runner_death_cure: {
+      title: 'Maze Runner: The Death Cure',
+      language: 'English',
+      author: 'James Dashner',
+      published: 2018,
+    },
+    fifty_shades_freed: {
+      title: 'Fifty Shades Freed',
+      language: 'English',
+      author: 'E L James',
+      published: 2018,
+    },
+    annihilation: {
+      title: 'Annihilation',
+      language: 'English',
+      author: 'Jeff VanderMeer',
+      published: 2018,
+    },
+    war_with_grandpa: {
+      title: 'The War with Grandpa',
+      language: 'English',
+      author: 'Robert Kimmel Smith',
+      published: 2018,
+    },
+    ready_player_one: {
+      title: 'Ready Player One',
+      language: 'English',
+      author: 'Ernest Cline',
+      published: 2018,
+    },
+    darkest_minds: {
+      title: 'The Darkest Minds',
+      language: 'English',
+      author: 'Alexandra Bracken',
+      published: 2018,
+    },
+    boy_erased: {
+      title: 'Boy Erased',
+      language: 'English',
+      author: 'Garrard Conley',
+      published: 2018,
+    },
+    first_man: {
+      title: 'First Man',
+      language: 'English',
+      author: 'James R. Hansen',
+      published: 2016,
+    },
+    little_women: {
+      title: 'Little Women',
+      language: 'English',
+      author: 'Louisa May Alcott',
+      published: 2018,
+    },
+    ophelia: {
+      title: 'Ophelia',
+      language: 'English',
+      author: 'Lisa Klein',
+      published: 2018,
+    },
+  };
+
+  /* cSpell:enable */
+
+  // 1.7 Find and download book covers for each book and construct a new object which has as keys the book IDs again,
+  //  and as value the path to the image source.
+  const bookImages = {
+    maze_runner_death_cure: 'images/maze_runner_death_cure.jpg',
+    fifty_shades_freed: 'images/fifty_shades_freed.jpg',
+    annihilation: 'images/annihilation.jpg',
+    war_with_grandpa: 'images/war_with_grandpa.jpg',
+    ready_player_one: 'images/ready_player_one.jpg',
+    darkest_minds: 'images/darkest_minds.jpg',
+    boy_erased: 'images/boy_erased.jpg',
+    first_man: 'images/first_man.jpg',
+    little_women: 'images/little_women.jpg',
+    ophelia: 'images/ophelia.jpg',
+  };
+
+  // 1.6 Beautify your html page with css (use the style.css file for that), add sources and alts to each of the images.
+
+  // 1.5 - 1.8
+  function bookDetails(booksList) {
+    for (let i = 0; i < bookTitles.length; i++) {
+      const listItem = document.createElement('li');
```

Comment:
Personally, I would have broken this part up into one or more functions, and then called the functions within the look passing in the book data. 
This works, it is just a maintainability issue, and personal preference.

Reply:
Many thanks, I'll take your feedback into consideration while improving this homework and next works.